view/script/patient.js

```diff
@@ -47,5 +50,53 @@
         events: window.events
     });
 
+    perWeekDatesConfig = datesByWeek(allDatesConfig);
+    config.data = perWeekDatesConfig;
+
+    $('.fc-today-button').click(function todayButtonClick () {
+        perWeekDatesConfig = datesByWeek(allDatesConfig);
+        config.data = perWeekDatesConfig;
+        new Chart(ctx, config);
+    });
+
+    $('.fc-prev-button').click(function prevButtonClick () {
+        perWeekDatesConfig = datesByWeek(allDatesConfig);
+        config.data = perWeekDatesConfig;
+        new Chart(ctx, config);
+    });
+
+    $('.fc-next-button').click(function nextButtonClick () {
+        perWeekDatesConfig = datesByWeek(allDatesConfig);
+        config.data = perWeekDatesConfig;
+        new Chart(ctx, config);
+    });
+
+    function datesByWeek (configData) {
+        var datesConfig = jQuery.extend(true, {}, configData);
+        var view = $('#calendar').fullCalendar('getView');
+        var datesPerView = [];
+        var timeLeftData = [];
+
+        var weekStartDate = new Date(view.intervalStart).getTime();
+        var weekEndDate = new Date(view.intervalEnd).getTime();
+        var index = 0;
+        var dateLabel = '';
+        var dateDatasetData = '';
+
+        for (index = 0; index < datesConfig.labels.length; index += 1) {
```

Comment:
using `Array.filter` could clean up this loop and (I think) could eliminate the need to deep clone `window.dates`

reference https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter